store/store.go

```diff
@@ -1534,7 +1534,13 @@ func downloadImpl(ctx context.Context, name, sha3_384, downloadURL string, user
 			}
 			break
 		}
-
+		if resume > 0 && resp.StatusCode != 206 {
+			logger.Debugf("server does not support resume")
+			resp.Body.Close()
+			w.Seek(0, os.SEEK_SET)
+			resume = 0
+			continue
```

Comment:
hmm I wonder if when this happens we should only end up making one request? When this happens in the wild, do servers first respond with a valid 206 Range request (i.e. giving only some of the data) on the first request, then respond with a 200 and the full data on subsequent requests? That seems a lot like a HTTP server bug that it only sometimes honors Range requests, but if that is what we see then this looks like the correct thing to do. 

OTOH, if we are receiving the full data in the 200 response, should we perhaps just detect that and not issue another request for the full data? Then we could do something like:

```
		if resume > 0 && resp.StatusCode != 206 {
			logger.Debugf("server does not support resume")
			if _, err := w.Seek(0, os.SEEK_SET); err != nil {
				return err
			}
                        h := crypto.SHA3_384.New()
			resume = 0
		}
```

Reply:
hmm actually thinking about it, if they respond with a 200, then shouldn't that response always have the full data and we can just resume without restarting like we do here?

Reply:
Excellent point, I think you are right and we don't need the retry. I will update the code soon.